library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -173,6 +233,72 @@ class Client() {
         }
     }
 
+    private suspend fun ffiXmtpClient(
+        options: ClientOptions?,
+        account: SigningKey?,
+        accountAddress: String,
+        appContext: Context?,
+        privateKeyBundleV1: PrivateKeyBundleV1,
+        legacyIdentitySource: LegacyIdentitySource,
+    ): FfiXmtpClient? {
+        val v3Client: FfiXmtpClient? =
+            if (options != null && options.enableLibXmtpV3 && options.appContext != null) {
+                val alias = "xmtp-${options.api.env}-${accountAddress.lowercase()}"
+
+                val dbDir = File(appContext?.filesDir?.absolutePath, "xmtp_db")
+                dbDir.mkdir()
+                val dbPath: String = dbDir.absolutePath + "/$alias.db3"
+
+                val keyStore = KeyStore.getInstance("AndroidKeyStore")
+                withContext(Dispatchers.IO) {
+                    keyStore.load(null)
+                }
+
+                val entry = keyStore.getEntry(alias, null)
+
+
+                val retrievedKey: SecretKey = if (entry is KeyStore.SecretKeyEntry) {
+                    entry.secretKey
+                } else {
+                    val keyGenerator =
+                        KeyGenerator.getInstance(KeyProperties.KEY_ALGORITHM_AES, "AndroidKeyStore")
+                    val keyGenParameterSpec = KeyGenParameterSpec.Builder(
+                        alias,
+                        KeyProperties.PURPOSE_ENCRYPT or KeyProperties.PURPOSE_DECRYPT
+                    ).setBlockModes(KeyProperties.BLOCK_MODE_GCM)
+                        .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_NONE)
+                        .setKeySize(256)
+                        .build()
+
+                    keyGenerator.init(keyGenParameterSpec)
+                    keyGenerator.generateKey()
+                }
+
+                createClient(
+                    logger = logger,
+                    host = "http://10.0.2.2:5556",
+                    isSecure = false,
+                    db = dbPath,
+                    encryptionKey = retrievedKey.encoded,
+                    accountAddress = accountAddress.lowercase(),
+                    legacyIdentitySource = legacyIdentitySource,
+                    legacySignedPrivateKeyProto = privateKeyBundleV1.toV2().identityKey.toByteArray()
+                )
+            } else {
+                null
+            }
+
+        if (v3Client?.textToSign() == null) {
+            v3Client?.registerIdentity(null)
+        } else {
+            v3Client.textToSign()?.let {
+                v3Client.registerIdentity(account?.sign(it))
```

Comment:
I think the `account` param of this function should be non-optional - if there is a text to be signed then we must request it, or the v3 client will be non-functional